src/pages/instances/actions/InstanceBulkActions.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { FC, Fragment, useState } from "react";
+import { updateInstanceBulkAction } from "api/instances";
+import { useNotify } from "context/notify";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import ConfirmationForce from "components/ConfirmationForce";
+import { LxdInstance, LxdInstanceAction } from "types/instance";
+import {
+  actionSummary,
+  instanceActions,
+  pluralizeInstance,
+} from "util/instanceBulkActions";
+import InstanceBulkAction from "pages/instances/actions/InstanceBulkAction";
+
+interface Props {
+  instances: LxdInstance[];
+  onStart: () => void;
+  onFinish: () => void;
+}
+
+const InstanceBulkActions: FC<Props> = ({ instances, onStart, onFinish }) => {
+  const notify = useNotify();
+  const queryClient = useQueryClient();
+  const [activeAction, setActiveAction] = useState<LxdInstanceAction | null>(
+    null
+  );
+  const [isForce, setForce] = useState(false);
+
+  const handleAction = (desiredAction: LxdInstanceAction) => {
+    setActiveAction(desiredAction);
+    onStart();
+    const actions = instanceActions(instances, desiredAction);
+
+    updateInstanceBulkAction(actions, isForce)
+      .then(() => {
+        const summary = actionSummary(actions);
+
+        notify.success(
+          summary.map(({ count, action }) => (
+            <Fragment key={action}>
+              <b>{count}</b> {pluralizeInstance(count)} {action}.<br />
+            </Fragment>
+          ))
```

Comment:
The "actions" here will always be only one kind of action, right? Or will there be cases when we can have different actions (with different outcomes) from the same bulk operation? I could not imagine any, but here it seems like we're treating this as an array of possibly different actions. Which edge cases am I failing to imagine here?

Reply:
Paused and stopped instances can be started
Running and paused instances can be stopped
Basically the `actionsToPerform` array with duplicate entries.

Reply:
We're talking about the summary here, right? Namely, the sentence that, at the of an operation, says for example:

> 2 instances stopped

So, my question is: is this an array? Can we have multiple lines out of it? Because that is what the code suggests. And I'm failing to imagine a case where you'd have multiple lines as the outcome of a single type of action.

Reply:
Right we changed this along the way and here we can only have one action in user's peek. Because unfreeze and start are the same thing for the user here.